crates/ra_assists/src/lib.rs

```diff
@@ -67,9 +86,14 @@ impl Assist {
     ///
     /// Assists are returned in the "unresolved" state, that is only labels are
     /// returned, without actual edits.
-    pub fn unresolved(db: &RootDatabase, config: &AssistConfig, range: FileRange) -> Vec<Assist> {
+    pub fn unresolved(
+        db: &RootDatabase,
+        config: &AssistConfig,
+        range: FileRange,
+        allowed: Option<Vec<AssistKind>>,
```

Comment:
Let's add this as a field of `AssistConfig`.

Reply:
I'm not sure that `AssistConfig` is the right place. That seems to be for global configuration while the filter is per request.

Reply:
So, originally we did use `Config` struct for global configuration. But than later I realised that global configuraiton is just a wrong pattern, and that per request configuration is just easier and more flexible. So all our `AsssitConfig`, `ComletionConfig`, etc, are scoped to a single method call, and it's fine to add filtering info there.